src/shared/utils/fluxExample.ts

```diff
@@ -0,0 +1,25 @@
+import {FluxFunction} from 'src/types/shared'
+
+export const getFluxExample = (func: FluxFunction) => {
+  const {fluxParameters = [], kind, name, package: packageName} = func
+
+  let example = `${packageName}.${name}`
+  if (kind.toLowerCase() === 'function') {
+    example =
+      `${packageName}.${name}(` +
+      fluxParameters.reduce((injectedParameters, parameter) => {
+        if (parameter.required) {
+          // add a comma if the current injected list is not all spaces
+          if (injectedParameters.trim().length !== 0) {
+            injectedParameters += ', '
+          }
+          injectedParameters += `${parameter.name}: `
+        } else {
+          injectedParameters += ' '
+        }
+        return injectedParameters
```

Comment:
if possible, don't mix concatenation and string interpolation - concatenation can be confused for the mathematical addition operator.

We could also reformulate this to have less syntax and fewer variable assignments (and thus be easier to follow):

```ts
if (parameter.required) {
  // add a comma if the current injected list is not all spaces
  if (injectedParameters.trim().length !== 0) {
    injectedParameters += ', '
  }
  return `${injectedParameters}${parameter.name}: `
}
return `${injectedParameters} `
```

This is a personal preference thing - I personally find it easier to set defaults and not deal with `else` clauses if possible.

I'd also argue that using `reduce` provides more overhead for not a lot of benefit, since we're overwriting strings anyway, which kind of defeats the purpose of using immutable functional paradigms.

```ts
let injectedParameters
for (const parameter of fluxParameters) {
  if (parameter.required) {
    // add a comma if the current injected list is not all spaces
    if (injectedParameters.trim().length !== 0) {
      injectedParameters += ', '
    }
    inectedParameters = `${injectedParameters}${parameter.name}: `
    continue
  }
  injectedParameters = `${injectedParameters} `
}

example = `${packageName}.${name}(${injectedParameters})`
```

but this is more of a preference thing that we don't have standards on. My personal preference is that `reduce` should be used to reduce a collection of values into a single value, not used to convert an array to an object or do string manipulation. Your choice on this comment, since it's a preference thing.

Reply:
This is a good observation. Sometimes, I have gotten so used to and familiar with certain utilities that I forget that not everyone may see the usage and understand immediately. Readability is better for maintenance.